### PR TITLE
added Proficency rules and glossary entries

### PR DESCRIPTION
**Pull request by bleehu**
To satisfy Issue #201

### CODE REVIEW

File: Glossary_of_Terms.txt [hunk 1]
```diff
@@ -34,6 +34,8 @@ RPG: Role-Playing Game, such as Dungeons and Dragons or Compound X!
 
 Player Character: The Player's avatar in the game world. A Player Character is made by following the procedures outlined in the CharacterCreation folder. Generally, a Player only controls one Character at a time; that Character is called a Player Character. 
 
+Proficiency: Compound X uses three types of Proficiencies Weapon Proficiency, Armor Proficiency, and Skill Proficiency. For Weapon or Armor Proficiency, see Basic Rules Assigning A Class To Your Character Section, subsections Weapon Proficiency or Armor Proficiency. For Skill Proficiency see Basic Rules RP/Skills section.
```
Comment (1sourcecontrol) [inline Glossary_of_Terms.txt]: Can you add some text that describes what a proficiency is? Like: "Proficiency: sufficient aptitude such that a character can use a given type of Weapon/Armor or use a Skill with higher capability." 

>Compound X uses three types of Proficiencies Weapon Proficiency, Armor Proficiency, and Skill Proficiency

Can you add a colon after "Proficiencies" or something here? There is a distinct pause but no punctuation.
And I think we really need to use quotes or something to indicate that we are referring to a section or chapter. For example, see Basic Rules doc.

>For Skill Proficiency see Basic Rules RP/Skills section

I think we need a comma after "Proficiency," it helps readability because of the pause from the prepositional phrase. 

>see Basic Rules Assigning A Class To Your Character Section 

I think we need to get in the habit (myself included) of putting quotes around titles/sections or all caps or _anything_ so that it's easier to read. Example:
"... see Basic Rules "Assigning A Class To Your Character" section, subsections "Weapon Proficiency" or..."
or better yet, what I've been trying to do (due to the weird pauses in the quotation) is as follows:
"... see section in the Basic Rules "Assigning A Class to Your Character," subsections "Weapon Proficiency"..."

File: Basic Rules.txt [hunk 2]
```diff
@@ -369,7 +369,31 @@ level 15. After level 15, the limit for stat points is 20. The number of
 initial base stat points that you have to apply is listed with your Race or 
 Species. See chapter 1: Races if you haven't yet. 
 
-Leveling Up:
+==== Assigning A Class to your Character ====
+	Most Characters start at level 1 so the Player will choose a Class for that 
+Character at that time. A Character will only ever have one Class at a time and
+under normal circumstances cannot change their class once it is picked.
```
Comment (1sourcecontrol) [inline Basic Rules.txt]: >A Character will only ever have one Class at a time and under normal circumstances cannot change their class once it is picked. 

Can you add comma(s) or parenthesis around, "under normal circumstances"? I believe this is a prepositional phrase that isn't essential to the meaning of the sentence. There is also a distinct pause after, "circumstances" that has no punctuation.

File: Basic Rules.txt [hunk 3]
```diff
@@ -369,7 +369,31 @@ level 15. After level 15, the limit for stat points is 20. The number of
 initial base stat points that you have to apply is listed with your Race or 
 Species. See chapter 1: Races if you haven't yet. 
 
-Leveling Up:
+==== Assigning A Class to your Character ====
+	Most Characters start at level 1 so the Player will choose a Class for that 
+Character at that time. A Character will only ever have one Class at a time and
+under normal circumstances cannot change their class once it is picked. 
+	On occasion especially for new Players, a character may start at level 0
```
Comment (1sourcecontrol) [inline Basic Rules.txt]: Also here, I think we need a comma after "occasion"

File: Basic Rules.txt [hunk 4]
```diff
@@ -369,7 +369,31 @@ level 15. After level 15, the limit for stat points is 20. The number of
 initial base stat points that you have to apply is listed with your Race or 
 Species. See chapter 1: Races if you haven't yet. 
 
-Leveling Up:
+==== Assigning A Class to your Character ====
+	Most Characters start at level 1 so the Player will choose a Class for that 
+Character at that time. A Character will only ever have one Class at a time and
+under normal circumstances cannot change their class once it is picked. 
+	On occasion especially for new Players, a character may start at level 0 
+with no Class. This is usually for a tutorial mission and a Class is selected
```
Comment (1sourcecontrol) [inline Basic Rules.txt]: Missing a comma before "and." The comma is necessary because you are separating two complete sentences with a coordinating conjunction.

**Review by 1sourcecontrol — CHANGES_REQUESTED** [review]
Thanks for adding this! Looks good, besides a few tweaks here and there.

File: Basic Rules.txt (hunk 2 above)
Comment (bleehu) [reply Basic Rules.txt]: wilco

File: Basic Rules.txt (hunk 3 above)
Comment (bleehu) [reply Basic Rules.txt]: wilco. I'll add parens around the "especially" clause as well.

File: Basic Rules.txt (hunk 4 above)
Comment (bleehu) [reply Basic Rules.txt]: wilco

File: Glossary_of_Terms.txt (hunk 1 above)
Comment (bleehu) [reply Glossary_of_Terms.txt]: wilco. And we should add something to that effect to the style guide, if we haven't already.

Comment (1sourcecontrol) [reply Glossary_of_Terms.txt]: Thanks for putting up with my nitpicks! :1st_place_medal: 
I added a PR for it

**Comment by bleehu** [comment]
All points implemented. Fresh review requested.

File: Basic Rules.txt [hunk 5]
```diff
@@ -359,17 +359,42 @@ determined by your character class and an associated primary stat. Your class ma
...
+Additionally, even after race/species bonuses, feats, equipment, and base stat 
 bonuses from leveling, no stat can ever be over 15 until your character is 
 level 15. After level 15, the limit for stat points is 20. The number of 
 initial base stat points that you have to apply is listed with your Race or 
 Species. See chapter 1: Races if you haven't yet. 
 
+===== Assigning A Class to your Character =====
+	Most Characters start at level 1 so the Player will choose a Class for that 
+Character at that time. A Character will only ever have one Class at a time, and
+under normal circumstances, cannot change their class once it is picked.
```
Comment (Turtlelord26) [inline Basic Rules.txt:377]: I don't think I agree with @1sourcecontrol about the comma after 'circumstances'. I don't see a pause as necessary here (if I were speaking it I wouldn't use one.), and it hurts the flow of the compounded sentence.

That said, I'll understand if you'd rather not revert.

File: Basic Rules.txt [hunk 6]
```diff
@@ -359,17 +359,42 @@ determined by your character class and an associated primary stat. Your class ma
...
+	Most Characters start at level 1 so the Player will choose a Class for that 
+Character at that time. A Character will only ever have one Class at a time, and
+under normal circumstances, cannot change their class once it is picked. 
+	On occasion (especially for new Players), a character may start at level 0 
+with no Class. This is usually for a tutorial mission and a Class is selected
+when the Character has completed the tutorial and leveled up to level 1. 
+
+===== Weapon Proficiencies =====
+	Each Class comes with a specific list of Weapon Proficiencies. If your 
+Character has a proficiency with a type of weapon, then they can use that weapon
```
Comment (Turtlelord26) [inline Basic Rules.txt:384]: If Proficiency is a glossaried term, then I believe our style mandates we capitalize all uses of it (as a game term)?

File: Basic Rules.txt [hunk 7]
```diff
@@ -359,17 +359,42 @@ determined by your character class and an associated primary stat. Your class ma
...
+in combat and can purchase weapons of that type in the world or during character
+creation. If your Character lacks proficiency with a weapon, then they can't 
+purchase or use a weapon of that type. 
+	Note that Weapon Proficiency is different than having Skills with that
```
Comment (Turtlelord26) [inline Basic Rules.txt:388]: Skill, singular. Alternatively, "Skills with weapons", with weapons plural.

File: Basic Rules.txt [hunk 8]
```diff
@@ -359,17 +359,42 @@ determined by your character class and an associated primary stat. Your class ma
...
+Character has a proficiency with a type of weapon, then they can use that weapon
+in combat and can purchase weapons of that type in the world or during character
+creation. If your Character lacks proficiency with a weapon, then they can't 
+purchase or use a weapon of that type. 
+	Note that Weapon Proficiency is different than having Skills with that 
+weapon. For more on Skills, see the RP/Skills section of this document.
+
+===== Armor Proficiencies =====
+	Just like each Class comes with a set of Weapon Proficiencies, it also comes
+with a list of Armor Proficiencies. Just like Weapon Proficiencies, if your
```
Comment (Turtlelord26) [inline Basic Rules.txt:393]: This is really heavily into writing style editing, but the double usage of "just like ... weapon proficiencies" is repetitive.

**Review by Turtlelord26 — CHANGES_REQUESTED** [review]
Just grammar and style. Nothing earth-shattering.

File: Basic Rules.txt (hunk 5 above)
Comment (Turtlelord26) [reply Basic Rules.txt:377]: Oh, duh, this isn't actually a compound sentence. Easy possible fix: change to "...circumstances, they cannot change..." and make it one.

File: Glossary_of_Terms.txt [hunk 9]
```diff
@@ -1,48 +1,117 @@
...
+
+DM: 
+    Dungeon Master. See Game Master. 
+
+Feat: 
+    As your character levels up, they get special perks or bonuses. These are 
+called Feats and are detailed in the CharacterCreation/03_Feats.txt document. 
+
+Feats, Class: 
+    See Class Feats
```
Comment (1sourcecontrol) [inline Glossary_of_Terms.txt:52]: Needs a period.

File: Glossary_of_Terms.txt [hunk 10]
```diff
@@ -1,48 +1,117 @@
...
+
+Game Master: 
+    Usually abbreviated as GM, the Game Master runs the game for the Players. 
+They are responsible for telling the story, keeping the game moving and deciding 
+how the NPCs behave and act. 
+
+GM: 
+See Game Master.
```
Comment (1sourcecontrol) [inline Glossary_of_Terms.txt:60]: Missing an indent.

File: Glossary_of_Terms.txt [hunk 11]
```diff
@@ -1,48 +1,117 @@
...
+circumstances). See topic "Opportunity Attack" under "Melee Combat" in the basic 
+rules.
+
+Player: 
+    A human in the real world who is playing the game. One Player may have many 
+Characters and might play in more than one game. Usually one Player plays one 
+Player Character at a time. 
+
+RPG: 
+Role-Playing Game, such as Dungeons and Dragons or Compound X!
```
Comment (1sourcecontrol) [inline Glossary_of_Terms.txt:85]: Missing an Indent.

File: Glossary_of_Terms.txt [hunk 12]
```diff
@@ -1,48 +1,117 @@
...
+
+Player Character: 
+    The Player's avatar in the game world. A Player Character is made by 
+following the procedures outlined in the CharacterCreation folder. Generally, a 
+Player only controls one Character at a time; that Character is called a Player 
+Character. 
+    In brief, Proficiency proficiency is a sufficient aptitude such that a
```
Comment (1sourcecontrol) [inline Glossary_of_Terms.txt:92]: >Proficiency proficiency

**Review by 1sourcecontrol — CHANGES_REQUESTED** [review]

**Comment by 1sourcecontrol** [comment]
Almost there! :dancer: :D

**Comment by 1sourcecontrol** [comment]
@bleehu I'm concerned with the large quantity of ignored fixes. Did you merge because they were scope creeps/ what's your plan of attack?

**Comment by bleehu** [comment]
ah, nope. I think what happened was I had a branch off of this one that I was expecting to take longer to get to dev than this one, but the work in the subsequent branch depended on the creation of the glossary. It was approved first and ended up squashing this one into dev with it. Which is why it names the commit, not the branch as being merged. 
TL;DR, I messed up. Look for a new PR with the fixes imminent.

**Comment by 1sourcecontrol** [comment]
Ah, I didn't notice that the commit was merged, and not the whole PR. That's confusing lol
Thanks!
